dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/filter-fee-record-payment-entity-groups.test.ts

```diff
@@ -0,0 +1,277 @@
+import { CURRENCY, FeeRecordEntityMockBuilder, PaymentEntityMockBuilder } from '@ukef/dtfs2-common';
+import { aUtilisationReport, getSqlIdGenerator } from '../../../../../../test-helpers';
+import { FeeRecordPaymentEntityGroup } from '../../../../../types/fee-record-payment-entity-group';
+import { filterFeeRecordPaymentEntityGroups } from './filter-fee-record-payment-entity-groups';
+import { ValidatedPaymentDetailsFilters } from '../../../../../types/utilisation-reports';
+
+describe('get-utilisation-report-reconciliation-details-by-id.controller helpers', () => {
+  const feeRecordIdGenerator = getSqlIdGenerator();
+
+  const aFeeRecord = () => FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).build();
+
+  const aFeeRecordWithFacilityId = (facilityId: string) =>
+    FeeRecordEntityMockBuilder.forReport(aUtilisationReport()).withId(feeRecordIdGenerator.next().value).withFacilityId(facilityId).build();
+
+  describe('filterFeeRecordPaymentEntityGroups', () => {
+    describe('facility id filter', () => {
+      it('returns the group where the supplied facility id filter matches one of the fee records in a group', () => {
```

Comment:
could you add should ....  we are going to have to standardise them all later but would be good to start on new tests.
Ideally a describe before each to make the it shorter would be good eg `describe('when the supplied facility id filter matches one of the fee records in a group', () => {`

Reply:
Tried to address this across all of the new tests